.github/workflows/publish-npm.yml

```diff
@@ -0,0 +1,62 @@
+name: Publish to NPM
+
+on:
+  release:
+    types: [published]
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: "21"
+          registry-url: "https://registry.npmjs.org"
+
+      - name: Install Rust
+        run: |
+          curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+          source $HOME/.cargo/env
+
+      - name: Install Solana CLI
+        run: |
+          sh -c "$(curl -sSfL https://release.solana.com/v1.18.15/install)"
+          export PATH="/home/runner/.local/share/solana/install/active_release/bin:$PATH"
+          solana --version
+
+      - name: Install Anchor CLI
+        run: |
+          cargo install --git https://github.com/coral-xyz/anchor --tag v0.30.0 anchor-cli --locked
+          anchor --version
+
+      - name: Build
+        run: anchor build
+
+      - name: Prepare IDL Files
+        run: |
+          mv target/idl ./
```

Comment:
nit: consider changing the build target path, by modifying `Cargo.toml`:
```
idl-build = ["my/idl/target/folder"]
```

If we want to generate the build in the root instead of target/idl always, this should be the way.

Reply:
If I do:

```toml
idl-build = ["foo/bar"]
```

I get:

```
anchor build
[2024-09-02T02:54:41.035855000Z ERROR cargo_build_sbf] Failed to obtain package metadata: `cargo metadata` exited with an error: error: failed to parse manifest at `/Users/fadeev/protocol-contracts-solana/programs/protocol-contracts-solana/Cargo.toml`
    
    Caused by:
      feature `idl-build` includes `foo/bar`, but `foo` is not a dependency
```

Seems like this array defines a list of dependencies, not an out path.

Reply:
Let's leave it for now, but this should work for anchor 30.